From 507cc575333ea0dad4fee756f072a95a09836c39 Mon Sep 17 00:00:00 2001
From: Barry Wardell <barry.wardell@gmail.com>
Date: Wed, 17 Aug 2011 23:51:41 +0200
Subject: [PATCH] Don't modify the OptionList when building. It was being
 silently modified based on e.g., the --debug, --optimise
 and --profile options to the build command. This could lead
 to very unexpected behaviour.

---
 lib/sim-build.py |   17 -----------------
 1 files changed, 0 insertions(+), 17 deletions(-)

diff --git a/lib/sim-build.py b/lib/sim-build.py
index 7db3188..7589377 100755
--- a/lib/sim-build.py
+++ b/lib/sim-build.py
@@ -120,23 +120,6 @@ def PrepConfiguration(configName):
     else:
         optionSettings = simlib.GetFileContents(optionlist)
     
-    DefineDatabase.AddReplacement("DEBUG", simlib.BoolToYesNo(build_debug))
-    DefineDatabase.AddReplacement("OPTIMISE", simlib.BoolToYesNo(build_optimise))
-    DefineDatabase.AddReplacement("UNSAFE", simlib.BoolToYesNo(build_unsafe))
-    DefineDatabase.AddReplacement("PROFILE", simlib.BoolToYesNo(build_profile))
-    
-    
-    # add support for CACHELINE_BYTES and CACHE_SIZE
-    CACHELINE_BYTES = simenv.LocalMachineEntry.GetKey("L3linesize")
-    
-    if CACHELINE_BYTES != None:
-        DefineDatabase.AddReplacement("CACHELINE_BYTES", CACHELINE_BYTES)
-    
-    CACHE_SIZE = simenv.LocalMachineEntry.GetKey("L3size")
-
-    if CACHE_SIZE != None:
-        DefineDatabase.AddReplacement("CACHE_SIZE", CACHE_SIZE)
-    
     optionSettings = DefineDatabase.SubAll(optionSettings)
 
     hasOutdatedConfig = (not hasCompleteConfig) or (storedOptionSettings == None) or (storedOptionSettings != optionSettings)
-- 
1.7.5.1

